src/graph.cc

```diff
@@ -554,24 +554,27 @@ bool ImplicitDepLoader::LoadDepFile(Edge* edge, const string& path,
     return false;
   }
 
-  // Preallocate space in edge->inputs_ to be filled in below.
-  vector<Node*>::iterator implicit_dep =
-      PreallocateSpace(edge, depfile.ins_.size());
-
-  // Add all its in-edges.
   for (vector<StringPiece>::iterator i = depfile.ins_.begin();
-       i != depfile.ins_.end(); ++i, ++implicit_dep) {
+       i != depfile.ins_.end(); ++i) {
     uint64_t slash_bits;
     if (!CanonicalizePath(const_cast<char*>(i->str_), &i->len_, &slash_bits,
                           err))
       return false;
 
     Node* node = state_->GetNode(*i, slash_bits);
-    *implicit_dep = node;
-    node->AddOutEdge(edge);
-    CreatePhonyInEdge(node);
+    depfile_deps->push_back(node);
   }
+  return true;
+}
 
+bool ImplicitDepLoader::LoadDepFile(Edge* edge, const string& path,
```

Comment:
Is it practical for this string to instead be StringPiece?

Reply:
I don't think it would be useful, there is a grand total of two call-sites for this and both have a std::string instance at the ready.

Reply:
Alright.